### PR TITLE
Do not use the fstab parameter on openbsd for mounting

### LINKED ISSUE #5591: mount module can't update flags on a mounted FS on OpenBSD
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

mount module

##### ANSIBLE VERSION

```
ansible 2.2.0.0
```

##### OS / ENVIRONMENT

OpenBSD

##### SUMMARY

https://github.com/ansible/ansible-modules-core/blob/devel/system/mount.py#L319 uses `-o remount`, which doesn't exist on OpenBSD. if you try to update the flags for an existing mountpoint, it will update fstab correctly, but then will try to remount/update the flags on the living FS, and mount module will fail.
##### STEPS TO REPRODUCE

```
#grep local /etc/fstab                                                                                       
/dev/wd0g /usr/local ffs rw,wxallowed 0 0

#ansible master -m mount -a 'name=/usr/local opts=rw,wxallowed,nosuid state=mounted fstype=ffs src=/dev/wd0g'
master | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Error mounting /usr/local: mount_ffs: -o remount: option not supported\n"
}
#grep local /etc/fstab                                                                                       
/dev/wd0g /usr/local ffs rw,wxallowed,nosuid 0 0
#mount |grep /usr/local
/dev/wd0g on /usr/local type ffs (local, wxallowed)
```

##### EXPECTED RESULTS

the living FS should have seen its opts updated.
##### ACTUAL RESULTS

mount module failed...  a possible fix would be to use `mount -u` which is more or less the equivalent of `mount -o remount`

```
#mount -u -o nosuid /usr/local
#mount |grep /usr/local       
/dev/wd0g on /usr/local type ffs (local, nosuid, wxallowed)

```

### PR DESCRIPTION
**Pull request by abadger**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

mount.py

##### ANSIBLE VERSION
```
devel
```

##### SUMMARY

OpenBSD's mount command doesn't allow selecting which fstab file to use.
So if we're operating on the live filesystem (mount or remount) return
an error if the user specified an fstab file.

Fixes #5591

### CODE REVIEW

**Comment by abadger** [comment]
@landryb here's the PR I promised you for the fstab parameter on OpenBSD.  See if it works for you.  I'm tempted to push it to devel but not to 2.2... I might just hold off on pushing it to 2.2.1 but queue it up for 2.2.2 if there aren't issues.

**Comment by gregdek** [comment]
Thanks @abadger for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by landryb** [comment]
Well, with mount.py taken from your pr on top of 2.2.0, i get the 'Do not specify the fstab parameter for OpenBSD hosts' error message but... i didnt specify one.
```
doas ansible -c local localhost -m mount -a 'name=/home opts=rw,dev,nosuid,wxallowed state=mounted fstype=ffs src=/dev/sd0d dump=1 passno=2'
```

File: system/mount.py [hunk 1]
```diff
@@ -328,7 +330,13 @@ def mount(module, args):
     if ismount(name):
         return remount(module, mount_bin, args)
 
-    cmd += _set_fstab_args(args)
+    if get_platform().lower() == 'openbsd':
+        # Use module.params['fstab'] here as args['fstab'] has been set to the
+        # default value.
+        if module.params['fstab'] is None:
```
Comment (landryb) [inline system/mount.py]: Should be inverted ?

File: system/mount.py [hunk 2]
```diff
@@ -364,7 +372,13 @@ def remount(module, mount_bin, args):
     else:
         cmd += ['-o', 'remount' ]
 
-    cmd += _set_fstab_args(args)
+    if get_platform().lower() == 'openbsd':
+        # Use module.params['fstab'] here as args['fstab'] has been set to the
+        # default value.
+        if module.params['fstab'] is None:
```
Comment (landryb) [inline system/mount.py]: Same thing, invert the condition ?

**Comment by landryb** [comment]
If i invert the logic (ie 'is not None' instead of 'is None'), i dont have the error message, but then the module still fails because it tries to umount a busy device. Still not ideal :)

**Comment by abadger** [comment]
Thanks.  Logic inversion fixed.

**Comment by abadger** [comment]
Merged to devel and pushed to temp-post-2.2.1-staging for inclusion in 2.2.2.

**Pull request merged**